DigitalLearningSolutions.Web/ViewModels/Common/AdminRolesViewModel.cs

```diff
@@ -0,0 +1,31 @@
+﻿namespace DigitalLearningSolutions.Web.ViewModels.Common
+{
+    using System.Collections.Generic;
+    using DigitalLearningSolutions.Data.Models.User;
+    using DigitalLearningSolutions.Web.ViewModels.Common.ViewComponents;
+    using DigitalLearningSolutions.Web.ViewModels.TrackingSystem.Centre.Administrator;
+    using Microsoft.AspNetCore.Mvc.Rendering;
+
+    public class AdminRolesViewModel : AdminRolesFormData
+    {
+        public readonly List<RadiosListItemViewModel> Radios = new List<RadiosListItemViewModel>();
+
+        public List<CheckboxListItemViewModel> Checkboxes = new List<CheckboxListItemViewModel>
+        {
+            AdminRoleInputs.CentreAdminCheckbox, AdminRoleInputs.SupervisorCheckbox
+        };
+
+        public AdminRolesViewModel(){}
+
+        public AdminRolesViewModel(User user, int centreId) : base(user)
+        {
+            CentreId = centreId;
+        }
+
+        public int CentreId { get; set; }
+        public IEnumerable<SelectListItem> LearningCategories { get; set; }
+
+        public bool NotAllRolesDisplayed => Radios.Count < 3 || Checkboxes.Count < 4;
+        public bool NoContentManagerOptionsAvailable => Radios.Count == 1;

```

Comment:
I'm not really sure whether there is, other than changing the numbers to named constants:
4 = Number of bool properties on this model
3 = Number of ContentManagementRoles in the enumeration
1 = Number of ContentManagementRoles that aren't actually a content management role???

Reply:
I would rather have named constants rather than magic numbers. I do feel like this can be determined at the SetUpCheckboxesAndRadioButtons stage but I'm not sure that would be an improvement over this.